Assets/Scripts/Crafting/CraftingRecipeUI.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Constants;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Crafting
+{
+    public class CraftingRecipeUI : MonoBehaviour
+    {
+        [Tooltip("The name of this recipe")] [SerializeField]
+        private TextMeshProUGUI txtTitle;
+
+        [Tooltip("The vertical layout group where the resources will be ordered in")] [SerializeField]
+        private VerticalLayoutGroup recipeResourceList;
+
+        [Tooltip("The background image to show if the recipe can be crafted")] [SerializeField]
+        private Image imgBackground;
+
+        [Tooltip("The button for crafting the item")] [SerializeField]
+        private Button craftButton;
+
+        [Tooltip("The crafting recipe resource prefab")] [SerializeField]
+        private CraftingRecipeResource craftingRecipeResourcePrefab;
+
+        /// <summary>
+        /// The crafting UI reference that holds the item handler
+        /// </summary>
+        private CraftingUI _craftingUI;
+
+        /// <summary>
+        /// The recipe of this CraftingRecipeUI element
+        /// </summary>
+        private CraftingRecipe _recipe;
+
+        private void Start() => OnItemUpdate();
+
+        /// <summary>
+        /// Instantiates all crafting recipe resources for the given recipe
+        /// </summary>
+        /// <param name="recipe">The recipe of this crafting recipe UI</param>
+        /// <param name="craftingUI">The crafting UI that should hold the correct <see cref="CraftingUI.ItemHandler"/></param>
+        public void InitRecipe(CraftingRecipe recipe, CraftingUI craftingUI)
+        {
+            _craftingUI = craftingUI;
+            _recipe = recipe;
+
+            txtTitle.SetText(_recipe.CreatedItemName);
+            foreach (CraftingRecipe.ResourceData resourceData in _recipe.NeededItems)
+                Instantiate(craftingRecipeResourcePrefab, recipeResourceList.transform)
```

Comment:
this is for displaying the things you need for crafting, right?

Reply:
Exactly, it spawns in each resource